docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -0,0 +1,416 @@
+  </tr>
+  
+  <tr>
+    <td>Dynamic Cluster Resource Info Cache</td>
+    <td>Distributed database cache to store the Cluster's available resource capacity. Since this information changes dynamically and will be collected periodically, there is no need to persist this information. Either ETCD Cache or Ignite satisfies our need for scalability, high availability, and cache performance<br><br>Each Global Scheduler partition's workload, geolocation tag, and profile tags are also saved in this DB cache</td>
+    <td>
+      <li>Distribute data store for scalability and high availability</li>
+      <li>Support cache data store for high performance</li>
+      <li>Support fast read operation</li>
+      <li>Support eventual data consistency</li>
+      <li>Notify object changes to API server which will notify the VM/Container Request Distributor</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>VM/Container Request Distributor</td>
+    <td>The Distributor tags each incoming VM/Container request based on its specification of geolocation/region/AZ and resource profile. It then assigns a Home Global Scheduler to it based on tag matching (each Global Scheduler is also tagged based on its assigned geolocations/regions/AZs and resource profiles supported by those geolocation/regions/AZs). If there are two or more matching Global Schedulers, the Distributor assigns a Global Scheduler that has the lowest load to it.</td>
+    <td>
+      <li>Register with API server to get notification of new or changes of VM/Container object</li>
+      <li>Distribute the VM/Container scheduling requests to an appropriate Home Global Scheduler</li>
+      <li>Support multiple partitions for scalability. If multiple partitions are needed, incoming requests from the API servers will be load balanced to those Distributor partitions. That is, each Distributor will be connected to a subset of API Servers and will handle a subset of incoming requests</li>
+      <li>Each Distributor is connected to all Global Schedulers and will distribute an incoming VM/Container request to a Home Global Scheduler based on tag matching</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>Dynamic Partition Mapper</td>
+    <td>The Mapper monitors the workload on each Global Scheduler and dynamically adjust each Global Scheduler's assigned  geolocations/regions/AZs and corresponding resource profile tags.</td>
+    <td>
+      <li>Determine the set of geolocations/regions/AZs each Global Scheduler is associated with</li>
+      <li>Construct the corresponding resource profile tags for each Global Scheduler</li>
+      <li>Monitor the workload of all the Global Schedulers</li>
+      <li>Dynamically adjust the assignment of geolocations/regions/AZs and corresponding profile tags to avoid overloading any single Global Scheduler</li>
+      <li>Update each Global Scheduler's workload, geolocation, resource profile tags in the DB cache</li>
+    </td>
+  </tr>
+  
+  <tr>
+    <td>Global Scheduler</td>
```

Comment:
Global scheduler: This interaction need cluster scheduler have the capability to accept scheduler request but not do the create VM task.